btree/btree.c

```diff
@@ -4055,7 +4102,10 @@ int64_t btree_iter_kv_tick(struct m0_sm_op *smop)
 			struct slot    s = {};
 			struct segaddr child;
 
+			lev = &oi->i_level[oi->i_pivot];
```

Comment:
you need not have to store sibling node at each level because we are only interested in leaf node sibling . If you are storing just to do node_put() as I see you are doing node_put() siblings present at each level in level_cleanup, you can do node_put at this state only, if it is internal node, you get the child index load child and then perform node_put for the parent .

Reply:
I am planning to modify the `get_tick` SLANT code also, to work in unlock mode. For that I will need sibling node. Just to have coherent code, I am storing l_sibling and clearing up it in level_cleanup(). We anyways had allocated memory for `l_sibling`, so assigning a node to this will not cause any performance degradation.

Reply:
okay